### PR TITLE
test: Add basic tests for zero/empty and nonzero/empty

### PR DESCRIPTION
**Pull request by gabrielsz**
* Add zero-ish value representation tests for int/uint/double/string/bool/null/bytes/list/map
* Add nonzero-ish value representation tests for int/uint/double/string/bool/bytes/list/map

### CODE REVIEW

File: tests/simple/testdata/basic.textproto [hunk 1]
```diff
@@ -9,10 +9,60 @@ section {
...
     expr: "''"
     value: { string_value: "" }
   }
+  test {
+    name: "self_eval_string_empty_quotes"
+    expr: '""'
+    value: { string_value: "" }
+  }
+  test {
+    name: "self_eval_string_r_prefix"
```
[the remark this answers is not in the document]
Comment (gabrielsz) [reply tests/simple/testdata/basic.textproto]: Thanks! Fixed

File: tests/simple/testdata/basic.textproto [hunk 2]
```diff
@@ -22,6 +72,62 @@ section {
...
+  }
+  test {
+    name: "self_eval_string_escape"
+    expr: "'\\''"
+    value: { string_value: "'" }
+  }
+  test {
+    name: "self_eval_binary_escape"
+    expr: "b'&yuml;'"
+    value: { bytes_value: "&yuml;" } # Shouldn't this escape?
```
[the remark this answers is not in the document]
Comment (gabrielsz) [reply tests/simple/testdata/basic.textproto]: Fixed. Thanks for the pointers!

I'll be leaving complicated examples for specific/specialized test suites =)
